File: 0058-length-of-last-word/0058-length-of-last-word.c
int lengthOfLastWord(char* s) {
    int length = 0, i;
    for(i = 0 ; s[i] != '\0' ; i++);

    while(i >= 0) {
        if((s[i] == ' ' && length == 0) || s[i] == '\0') {
            i--;
            continue;
        } else if(s[i] == ' ') return length;
        else length++;
        i--;
    }

    return length;
}

File: 0058-length-of-last-word/0058-length-of-last-word.java
class Solution {
    public int lengthOfLastWord(String s) {
        int length = 0;
        int i = s.length() - 1;

        while(i >= 0) {
            if(s.charAt(i) == ' ' && length == 0) {
                i--;
                continue;
            } else if(s.charAt(i) == ' ') return length;
            else length++;

            i--;
        }

        return length;
    }
}